### PR TITLE
atmega timer: Interrupt Pin

### PR DESCRIPTION
**Pull request by Josar**
Debug pin can be used to probe timer interrupts. Thus, determine when an interrupt occurs and how long the ISR takes.

The timer interrupt times can be probed with a oscilloscope if pin `DEBUG_TIMER_PIN`, the respective `DEBUG_TIMER_PORT` and `DDEBUG_TIMER_DDR` are defined in the makefile as follows.

```
CFLAGS += -DDEBUG_TIMER_PORT=PORTF
CFLAGS += -DDEBUG_TIMER_DDR=DDRF
CFLAGS += -DDEBUG_TIMER_PIN=PORTF4
```

### CODE REVIEW

**Comment by Josar** [comment]
@aabadie could you have a look?

File: cpu/atmega_common/periph/timer.c [hunk 1]
```diff
@@ -169,6 +193,10 @@ void timer_start(tim_t tim)
 #ifdef TIMER_NUMOF
 static inline void _isr(tim_t tim, int chan)
 {
+#if defined(DEBUG_TIMER_PORT)
+    DEBUG_TIMER_PORT |= (1 << DEBUG_TIMER_PIN);
+    DEBUG_TIMER_PORT &= ~(1 << DEBUG_TIMER_PIN);
```
[the remark this answers is not in the document]
Comment (Josar) [reply cpu/atmega_common/periph/timer.c]: @A-Paul thank you for that. As the context swap of the atmegas was changed and setting the pin to low was not executed in the right time, as the context got switched, i changed the debug pin to only signilizes the start of the timer interrupt. And forgott to update the documentation.

I change the Pin toogle to be executed before the context swap, thus the documentation fits the function, only that the context swap is not included in the high period anymore.

**Comment by Josar** [comment]
Squash and rebase?

File: cpu/atmega_common/periph/timer.c [hunk 2]
```diff
@@ -32,6 +32,22 @@
...
+ *
+ * The timer interrupt times can be probed with a oscilloscope if pin
+ * `DEBUG_TIMER_PIN`, the respective `DEBUG_TIMER_PORT` and `DDEBUG_TIMER_DDR`
+ * are defined in the makefile as follows.
+ *
+ * CFLAGS += -DDEBUG_TIMER_PORT=PORTF
+ * CFLAGS += -DDEBUG_TIMER_DDR=DDRF
+ * CFLAGS += -DDEBUG_TIMER_PIN=PORTF4
+ */
+#if defined(DEBUG_TIMER_PORT)
```
Comment (A-Paul) [inline cpu/atmega_common/periph/timer.c]: "board.h" is already included in line 25.

**Comment by PeterKietzmann** [comment]
@Josar  please rebase. @A-Paul comments addressed, mind to approve?

**Comment by Josar** [comment]
@PeterKietzmann done

**Comment by A-Paul** [comment]
I wiil do a HW test tomorrow.

**Comment by A-Paul** [comment]
Hi @Josar , I have built and run some of the *tests/xtimer_** on a *arduino-mega2560* and connected to an oscilloscope. Works like it is supposed.
However, I have gotten a little concerned, because this is a very particular approach for a functionality that might be helpful on a more general level.
@PeterKietzmann mentioned a PR (#7352) that has a similar direction. Do you mind having a look at it  and maybe combine your efforts with @haukepetersen?

**Comment by Josar** [comment]
@A-Paul i had a quick view over the approach from https://github.com/RIOT-OS/RIOT/pull/7352.

And yes it seems more generalized, but it is also on halt for quite a while.

One other thing is that if it is used like this it is not flexible anymore. Means if i use e.g. `DBGPIN1` for the timer and in a seperate debug atempt also use it for e.g. context swap then only one or both can be enabled ore both at the same time on the same pin. I can not choose the pin seperatly anymore.

This Approach defines a well defined PIN for a single action e.g. Timer, This can be freely set to a pin.
The same could then be done for context swap with e.g `-DDEBUG_CONTEXT_SWAP_PIN=PORTF5` and also the pin can be choosen freely.

I am not in favor of the generalized approach shown in https://github.com/RIOT-OS/RIOT/pull/7352 for a low level driver debug pin solution.

The generelized approach could be somthing which could be used in application level as shown here https://github.com/RIOT-OS/RIOT/pull/8865/files. 
This would increase pin toggle speed and would benefit from generalization as each platform could use the same test and configure the debug pin at the `debugpin_cpu.h`

**Comment by A-Paul** [comment]
@Josar thank you for you investigation. I will follow you reasoning and merge, but I have a last change request. I'll place it in a review comment.

**Comment by Josar** [comment]
@A-Paul done.

**Comment by A-Paul** [comment]
@Josar, thank you for addressing all change requests. CI is happy. ACK and go.

**Pull request merged**